symengine/ruby/lib/symengine/basic.rb

```diff
@@ -0,0 +1,13 @@
+require 'set'
+
+module SymEngine
+  class Basic
+    def free_symbols
+      pr_free_symbols.to_set
+    end
+
+    def hash
+      [self.to_s].hash
```

Comment:
Just write a C hash function that returns FIXNUM, and that checks that `sizeof(size_t) == sizeof(FIXNUM)`.